### PR TITLE
Remove gui libtorrent tests

### PR DESCRIPTION
**Pull request by devos50**
We are already testing whether a download of libtorrent works (see `API/test_download.py`).

Also, the video player is tested by `test_vod.py` and playback will be tested in the new GUI tests.

It might be that the code coverage drops but I will fix that in this PR.

UPDATE: I've written some tests for the libtorrent core download process. `VODFile` will be tested after the video server is refactored.

### CODE REVIEW

**Comment by devos50** [comment]
Unable to check the code coverage since the tools is broken :/

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
Amazing, no code coverage drop :D

@lfdversluis can you review?

**Comment by lfdversluis** [comment]
Looks good

**Comment by devos50** [comment]
@whirm can you review?

**Comment by whirm** [comment]
retest this please

**Comment by whirm** [comment]
@devos50 I've added total coverage reports too

**Comment by whirm** [comment]
retest this please

**Comment by whirm** [comment]
@devos50 you can compare with upstream now: https://jenkins.tribler.org/job/GH_Tribler_PR_coverage_check/2388/cobertura/

**Comment by devos50** [comment]
@whirm great, thanks! I will take a look.

**Comment by whirm** [comment]
@devos50 could you send the decorator fixes to `next` please?

**Comment by devos50** [comment]
@whirm see #2432

**Comment by devos50** [comment]
```
======================================================================
ERROR: Testing whether the API returns a 404 if no channel has been created when adding a rss feed
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Test/test_as_server.py", line 272, in tearDown
    self._shutdown_session(self.session)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Test/test_as_server.py", line 359, in _shutdown_session
    assert diff < waittime, "test_as_server: took too long for Session to shutdown"
AssertionError: test_as_server: took too long for Session to shutdown
-------------------- >> begin captured logging << --------------------
```

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
Tunnel test timeout...

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@lfdversluis can you review again (added a bunch of tests)?

File: Tribler/Test/Core/test_downloadstate.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        stats = {'up': 123, 'down': 1234, 'stats': mock_transferred, 'time': 42, 'vod_prebuf_frac_consec': 43,
+                 'vod_prebuf_frac': 44, 'vod': True, 'tracker_status': {'a': 'b'}}
+        self.assertEqual(download_state.get_current_speed(UPLOAD), 0)
+        self.assertEqual(download_state.get_total_transferred(UPLOAD), 0)
+        self.assertEqual(download_state.get_num_peers(), 0)
+        self.assertEqual(download_state.get_num_nonseeds(), 0)
+        self.assertEqual(download_state.get_num_seeds_peers(), (0, 0))
+        self.assertEqual(download_state.get_vod_prebuffering_progress_consec(), 0)
+        self.assertEqual(download_state.get_vod_prebuffering_progress(), 0)
+        download_state.status = DLSTATUS_STOPPED
```
Comment (lfdversluis) [inline Tribler/Test/Core/test_downloadstate.py]: can there be a test with DLSTATUS_SEEDING?

**Comment by lfdversluis** [comment]
`19: R0915: (too-many-statements), TestDownloadState.test_getters_setters: Too many statements (54/50)` can you resolve this one?

**Comment by lfdversluis** [comment]
Comments given

File: Tribler/Test/Core/test_downloadstate.py (hunk 1 above)
Comment (devos50) [inline Tribler/Test/Core/test_downloadstate.py]: There some logic that relies on `DLSTATUS_SEEDING` in the init method. I've added some small tests to also test this init method.

File: Tribler/Test/Core/Libtorrent/test_libtorrent_download_impl.py [hunk 2]
```diff
@@ -7,12 +7,13 @@
...
 from Tribler.Core.Utilities.configparser import CallbackConfigParser
 from Tribler.Core.Utilities.twisted_thread import deferred, reactor
+from Tribler.Test.Core.base_test import TriblerCoreTest, MockObject
 from Tribler.Test.test_as_server import TestAsServer, TESTS_DATA_DIR
 
 
 class TestLibtorrentDownloadImpl(TestAsServer):
     """
-    This class provides unit tests tha tes tthe LibtorrentDownloadImpl class.
+    This class provides unit tests tha test the LibtorrentDownloadImpl class.
```
Comment (whirm) [inline Tribler/Test/Core/Libtorrent/test_libtorrent_download_impl.py]: that

Comment (whirm) [inline Tribler/Test/Core/Libtorrent/test_libtorrent_download_impl.py]: Or even This class contains tests for....

**Comment by whirm** [comment]
Otherwise looks fine

**Comment by devos50** [comment]
@whirm ready again

**Pull request merged**